[blog] Add link to Base UI API changes RFC (#41089)
mui/material-ui: docs/pages/blog/base-ui-2024-plans.md

To address these issues, we're considering adopting a new API that would assign a discrete subcomponent to each DOM node—the pattern favored by many other headless component libraries (think: `<Slider.Track />`, `<Slider.Thumb />`, etc.).
This pattern has the potential to radically improve the customization experience, both for styles and structure.

> We're still fleshing out the details.

Reviewer: ```suggestion
We're still fleshing out the details, but we know
```

(then the line below continues). @samuelsycamore, how about this so it's more connected to the next sentence? It reads a bit more natural to me that way!

Author: Actually, these sentences could be in different paragraphs. The fact that we're working on the API details is a separate piece of information that that we want a smooth migration path.

What do you think about:

> We are still fleshing out the API and implementation details. If you have any questions or remarks about this change, chime in on [the RFC on GitHub](https://github.com/mui/material-ui/discussions/41085).
>
> We know that a significant number of projects depend on the existing API, and we want to assure you that one of our top priorities is to provide a smooth migration experience.

Reviewer: Sounds great to me! 👍

We know that a significant number of projects depend on the existing API, and we want to assure you that one of our top priorities is to provide a smooth migration experience.

> If you have any questions or remarks about this change, feel free to comment on the [RFC](https://github.com/mui/material-ui/discussions/41085) we recently published.

Reviewer: ```suggestion
If you have any questions or remarks about this change, chime in on [the recently published RFC](https://github.com/mui/material-ui/discussions/41085).
```

@samuelsycamore, how about this? Just an alternative. And a small pet peeve of mine is when links are on small words (such as "RFC" in this case) 😅 so maybe it's easier to click on it if we extend to the full mention?
